x/metadata/keeper/objectstore.go

```diff
@@ -8,60 +8,41 @@ import (
 	"github.com/provenance-io/provenance/x/metadata/types"
 )
 
-// Handler is a name record handler function for use with IterateRecords.
-type ObjectStoreHandler func(record types.ObjectStoreLocator) error
-
-func (k Keeper) GetOsLocatorRecord(ctx sdk.Context, ownerAddress sdk.AccAddress) (osLocator types.ObjectStoreLocator, found bool) {
-	key, err := types.GetOsLocatorAddressKeyPrefix(ownerAddress)
-	if err != nil {
-		return types.ObjectStoreLocator{}, false
-	}
+// GetOsLocatorRecord Gets the object store locator entry from the kvstore for the given owner address.
+func (k Keeper) GetOsLocatorRecord(ctx sdk.Context, ownerAddr sdk.AccAddress) (osLocator types.ObjectStoreLocator, found bool) {
+	key := types.GetOSLocatorKey(ownerAddr)
 	store := ctx.KVStore(k.storeKey)
 	b := store.Get(key)
 	if b == nil {
 		return types.ObjectStoreLocator{}, false
 	}
-	err = k.cdc.UnmarshalBinaryBare(b, &osLocator)
+	err := k.cdc.UnmarshalBinaryBare(b, &osLocator)
 	if err != nil {
-		ctx.Logger().Error("failed to get locator", "err", err)
+		ctx.Logger().Error("failed to unmarshal locator", "err", err)
 		return types.ObjectStoreLocator{}, false
 	}
 	return osLocator, true
 }
 
-// Logger returns a module-specific logger.
-func (k Keeper) OSLocatorExists(ctx sdk.Context, ownerAddr string) bool {
-	address, err := sdk.AccAddressFromBech32(ownerAddr)
-	if err != nil {
-		ctx.Logger().Error("failed to get locator", "err", err)
-		return false
-	}
-	key, err := types.GetOsLocatorAddressKeyPrefix(address)
-
-	if err != nil {
-		ctx.Logger().Error("failed to get locator", "err", err)
-		return false
-	}
+// OSLocatorExists checks if the provided bech32 owner address has a OSL entry in the kvstore.
+func (k Keeper) OSLocatorExists(ctx sdk.Context, ownerAddr sdk.AccAddress) bool {
+	key := types.GetOSLocatorKey(ownerAddr)
 	store := ctx.KVStore(k.storeKey)
-
 	return store.Has(key)
 }
 
-// SetOSLocatorRecord binds a oslocator to an address. An error is returned if no account exists for the address.
+// SetOSLocatorRecord binds an OS Locator to an address in the kvstore.
+// An error is returned if no account exists for the address.
+// An error is returned if an OS Locator already exists for the address.
 func (k Keeper) SetOSLocatorRecord(ctx sdk.Context, ownerAddr sdk.AccAddress, uri string) error {
-	var err error
-
 	urlToPersist, err := k.checkValidURI(uri, ctx)
 	if err != nil {
 		return err
 	}
 	if account := k.authKeeper.GetAccount(ctx, ownerAddr); account == nil {
```

Comment:
This check shouldn't be here ... the owner account would have to sign a request to set a locator in the first place and doing this extra account exists check in the keeper leads to methods like the import locator that genesis and test processes need to set locators without an explicit dependency.  Using the auth module to check if an account exists seems like an antipattern.